csharp/src/Google.Protobuf/MessageParser.cs

```diff
@@ -71,6 +71,21 @@ public IMessage ParseFrom(byte[] data)
         }
 
         /// <summary>
+        /// Parses a message from a byte array slice.
+        /// </summary>
+        /// <param name="data">The byte array containing the message. Must not be null.</param>
+        /// <param name="offset">The offset of the slice to parse.</param>
+        /// <param name="length">The length of the slice to parse.</param>
+        /// <returns>The newly parsed message.</returns>
+        public IMessage ParseFrom(byte[] data, int offset, int length)
+        {
+            ProtoPreconditions.CheckNotNull(data, "data");
```

Comment:
I'd ditch this given that it'll be validated in the call to message.MergeFrom.

(If you change the parameter names to buffer in both cases, you could rely on the CodedInputStream constructor validation, but I prefer data instead of buffer here...)